File: src/main/java/com/manicure/service/OrderService.java
package com.manicure.service;

import java.util.Map;

import com.manicure.entity.TbOrder;
import com.manicure.entity.TbPayLog;

import entity.PageResult;

public interface OrderService {

	/**
	 * 添加订单
	 * @param order
	 */
	public Map add(TbOrder order);
	
	/**
	 * 根据用户ID获取支付日志
	 * @param userId
	 * @return
	 */
	public TbPayLog searchPayLogFromRedis(String userId);
	
	
	/**
	 * 支付成功修改状态
	 * @param out_trade_no
	 * @param transaction_id
	 */
	public void updateOrderStatus(String id,String userName);

	/**
	 * 根据id和useid查找订单和订单项封装map对象
	 * @param id
	 * @param username
	 * @return
	 */
	public Map findById(String id, String username);
	
	/**
	 * 用户分页查看订单列表
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize,String username,Integer time);

	/**
	 * 管理员查询订单发货
	 * @param pageNum
	 * @param pageSize
	 * @param tbOrder
	 * @return
	 */
	public PageResult search(int pageNum,int pageSize,TbOrder tbOrder);


	/**
	 * 管理员更新状态
	 * @param ids
	 */
	public void updateStatus(String[] ids,String status);

	
}

File: src/main/webapp/js/service/orderService.js
//服务层
app.service('orderService',function($http){
	    	
	
	
	//根据id获取order和ordeItems  
	this.findById=function(id){
		return $http.get('../order/findById.do?id='+id);		
	};
	
	//查找订单
	this.findPage = function(page,rows,time){
		return $http.get('../order/findPage.do?page='+page+'&rows='+rows+'&time='+time);
	};

	//管理员查找订单

	this.search = function(page,rows,entity){
		return $http.post('../order/search.do?page='+page+'&rows='+rows,entity);
	};

	//更新状态
	this.updateStatus = function(ids,status){
		return $http.get('../order/updateStatus.do?ids='+ids+'&status='+status);
	}
});
